public/app/features/alerting/unified/components/receivers/ReceiversTable.tsx

```diff
@@ -0,0 +1,83 @@
+import { useStyles } from '@grafana/ui';
+import { receiverTypeNames } from 'app/plugins/datasource/alertmanager/consts';
+import { AlertManagerCortexConfig } from 'app/plugins/datasource/alertmanager/types';
+import { capitalize } from 'lodash';
+import React, { FC, useMemo } from 'react';
+import { useUnifiedAlertingSelector } from '../../hooks/useUnifiedAlertingSelector';
+import { getAlertTableStyles } from '../../styles/table';
+import { ActionButton } from '../rules/ActionButton';
+import { ActionIcon } from '../rules/ActionIcon';
+import { ReceiversSection } from './ReceiversSection';
+
+interface Props {
+  config: AlertManagerCortexConfig;
+}
+
+export const ReceiversTable: FC<Props> = ({ config }) => {
+  const tableStyles = useStyles(getAlertTableStyles);
+
+  const grafanaReceiverTypes = useUnifiedAlertingSelector((state) => state.grafanaReceiverTypes);
+
+  const receivers = useMemo(
+    () =>
+      config.alertmanager_config.receivers?.map((receiver) => ({
+        name: receiver.name,
+        types: [
+          ...(receiver.grafana_managed_receiver_configs?.map((recv) => recv.type) ?? []).map(
+            (type) => grafanaReceiverTypes.result?.find((r) => r.type === type)?.name ?? capitalize(type)
+          ),
+          ...Object.entries(receiver)
+            .filter(
+              ([key, value]) =>
+                key !== 'grafana_managed_receiver_configs' &&
+                key.endsWith('_configs') &&
+                Array.isArray(value) &&
+                !!value.length
+            )
+            .map(([key]) => key.replace('_configs', ''))
+            .map((type) => receiverTypeNames[type] ?? capitalize(type)),
+        ],
+      })) ?? [],
+    [config, grafanaReceiverTypes.result]
+  );
```

Comment:
nit: sorry to say, but this block needs some refactoring. It is suuper hard to read and argue about what is happening here.

Reply:
I call this function `jobSecurity()` :laughing:   
you're right though
- [ ] make it more readable

Reply:
Yea this looks very messy and hard to read, and code like this needs unit tests

Reply:
refactored this notifier name extraction logic, improved some namings and wrote tests for this component.